src/events/SDL_categories.c

```diff
@@ -0,0 +1,256 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+/* SDL event categories */
+
+#include "SDL_events_c.h"
+#include "SDL_categories_c.h"
+
+SDL_EventCategory SDL_GetEventCategory(Uint32 type) {
+    if (type >= SDL_EVENT_USER && type <= SDL_EVENT_LAST) {
+        return SDL_EVENTCATEGORY_USER;
+    }
+    switch (type) {
+    default:
+        return SDL_SetError("Unknown event type");
+
+    case SDL_EVENT_KEYMAP_CHANGED:
+    case SDL_EVENT_TERMINATING:
+    case SDL_EVENT_LOW_MEMORY:
+    case SDL_EVENT_WILL_ENTER_BACKGROUND:
+    case SDL_EVENT_DID_ENTER_BACKGROUND:
+    case SDL_EVENT_WILL_ENTER_FOREGROUND:
+    case SDL_EVENT_DID_ENTER_FOREGROUND:
+    case SDL_EVENT_LOCALE_CHANGED:
+    case SDL_EVENT_SYSTEM_THEME_CHANGED:
+    case SDL_EVENT_RENDER_TARGETS_RESET:
+    case SDL_EVENT_RENDER_DEVICE_RESET:
+        return SDL_EVENTCATEGORY_SYSTEM;
+
+    case SDL_EVENT_QUIT:
+        return SDL_EVENTCATEGORY_QUIT;
+
+    case SDL_EVENT_DISPLAY_ORIENTATION:
+    case SDL_EVENT_DISPLAY_ADDED:
+    case SDL_EVENT_DISPLAY_REMOVED:
+    case SDL_EVENT_DISPLAY_MOVED:
+    case SDL_EVENT_DISPLAY_DESKTOP_MODE_CHANGED:
+    case SDL_EVENT_DISPLAY_CURRENT_MODE_CHANGED:
+    case SDL_EVENT_DISPLAY_CONTENT_SCALE_CHANGED:
+        return SDL_EVENTCATEGORY_DISPLAY;
+
+    case SDL_EVENT_WINDOW_SHOWN:
```

Comment:
Should this also include sdl2-compat values, such as 0x200 for `SDL_WINDOWEVENT` and 0x201 for `SDL_EVENT_SYSWM`?

Reply:
No, it shouldn't.